### ISSUE TITLE
Unexpected Start code in second entry.

### ISSUE THREAD

**Issue by yunikkk** [gojuno/composer#15]
One run ended with the following :

```
java -jar ci/internal/composer.jar --apk ./app/build/outputs/apk/XXX.apk --test-apk app/build/outputs/apk/XXX-debug-androidTest.apk --test-package com.gojuno.driver.dev.test --test-runner android.support.test.runner.JunoAndroidRunner --output-directory artifacts/coffeetray-output --instrumentation-arguments retryUiTests true reportFailuresToCrashlytics true
[Tue Apr 11 09:41:22 UTC 2017]: 6 connected adb device(s): [AdbDevice(id=emulator-5564, online=true), AdbDevice(id=emulator-5562, online=true), AdbDevice(id=emulator-5560, online=true), AdbDevice(id=emulator-5558, online=true), AdbDevice(id=emulator-5556, online=true), AdbDevice(id=emulator-5554, online=true)]
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5564] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5562] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5560] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5558] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5556] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:22 UTC 2017]: [emulator-5554] Installing apk... pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:25 UTC 2017]: [emulator-5558] Successfully installed apk in 3 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:25 UTC 2017]: [emulator-5558] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:25 UTC 2017]: [emulator-5556] Successfully installed apk in 3 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:25 UTC 2017]: [emulator-5556] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:26 UTC 2017]: [emulator-5554] Successfully installed apk in 3 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:26 UTC 2017]: [emulator-5554] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:26 UTC 2017]: [emulator-5560] Successfully installed apk in 3 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:26 UTC 2017]: [emulator-5560] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:27 UTC 2017]: [emulator-5556] Successfully installed apk in 2 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:27 UTC 2017]: [emulator-5556] Starting tests...
[Tue Apr 11 09:41:28 UTC 2017]: [emulator-5554] Successfully installed apk in 2 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:28 UTC 2017]: [emulator-5554] Starting tests...
[Tue Apr 11 09:41:28 UTC 2017]: [emulator-5558] Successfully installed apk in 2 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:28 UTC 2017]: [emulator-5558] Starting tests...
[Tue Apr 11 09:41:29 UTC 2017]: [emulator-5560] Successfully installed apk in 2 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:29 UTC 2017]: [emulator-5560] Starting tests...
[Tue Apr 11 09:41:29 UTC 2017]: [emulator-5562] Successfully installed apk in 6 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:29 UTC 2017]: [emulator-5562] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:32 UTC 2017]: [emulator-5564] Successfully installed apk in 9 seconds, pathToApk = ./app/build/outputs/apk/XXX.apk
[Tue Apr 11 09:41:32 UTC 2017]: [emulator-5564] Installing apk... pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:33 UTC 2017]: [emulator-5562] Successfully installed apk in 4 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:33 UTC 2017]: [emulator-5562] Starting tests...
[Tue Apr 11 09:41:35 UTC 2017]: [emulator-5564] Successfully installed apk in 3 seconds, pathToApk = app/build/outputs/apk/XXX-debug-androidTest.apk
[Tue Apr 11 09:41:35 UTC 2017]: [emulator-5564] Starting tests...
[Tue Apr 11 09:41:41 UTC 2017]: [emulator-5556] Test passed in 8 seconds: .tests.menu.about.AboutUITest.testAboutVersionName
[Tue Apr 11 09:41:41 UTC 2017]: [emulator-5558] Test passed in 7 seconds: .tests.menu.account.AccountUINegativeTest.testDriverCarPlateIsNotDisplayed
[Tue Apr 11 09:41:42 UTC 2017]: [emulator-5554] Test passed in 8 seconds: .tests.menu.account.AccountUIPositiveTest.testDriverProfilePictureIsDisplayed
[Tue Apr 11 09:41:43 UTC 2017]: [emulator-5560] Test passed in 8 seconds: .tests.menu.MenuHeaderPositiveTest.testDriverRating
[Tue Apr 11 09:41:44 UTC 2017]: [emulator-5562] Error during tests run: java.lang.IllegalStateException: Unexpected Start code in second entry, please report that to Composer maintainers (InstrumentationEntry(numTests=11, stream=.tests.menu.MenuHeaderPositiveTest:, id=AndroidJUnitRunner, test=testDriverPicture, clazz=.tests.menu.MenuHeaderPositiveTest, current=1, stack=, statusCode=Start, timestampNanos=95440347915313), InstrumentationEntry(numTests=11, stream=.tests.menu.MenuHeaderPositiveTest:, id=AndroidJUnitRunner, test=testDriverPicture, clazz=.tests.menu.MenuHeaderPositiveTest, current=1, stack=, statusCode=Start, timestampNanos=95447432113000))
Exception in thread "main" java.lang.IllegalStateException: Unexpected Start code in second entry, please report that to Composer maintainers (InstrumentationEntry(numTests=11, stream=.tests.menu.MenuHeaderPositiveTest:, id=AndroidJUnitRunner, test=testDriverPicture, clazz=.tests.menu.MenuHeaderPositiveTest, current=1, stack=, statusCode=Start, timestampNanos=95440347915313), InstrumentationEntry(numTests=11, stream=.tests.menu.MenuHeaderPositiveTest:, id=AndroidJUnitRunner, test=testDriverPicture, clazz=.tests.menu.MenuHeaderPositiveTest, current=1, stack=, statusCode=Start, timestampNanos=95447432113000))
	at com.gojuno.composer.InstrumentationKt$asTests$1.call(Instrumentation.kt:119)
	at com.gojuno.composer.InstrumentationKt$asTests$1.call(Instrumentation.kt)
	at rx.internal.operators.OperatorScan$3.onNext(OperatorScan.java:138)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:395)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:355)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:846)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
	at rx.internal.operators.OnSubscribeFilter$FilterSubscriber.onNext(OnSubscribeFilter.java:76)
	at rx.internal.operators.OperatorScan$InitialProducer.emitLoop(OperatorScan.java:323)
	at rx.internal.operators.OperatorScan$InitialProducer.emit(OperatorScan.java:295)
	at rx.internal.operators.OperatorScan$InitialProducer.onNext(OperatorScan.java:202)
	at rx.internal.operators.OperatorScan$3.onNext(OperatorScan.java:144)
	at rx.internal.operators.OperatorTakeWhile$2.onNext(OperatorTakeWhile.java:67)
	at rx.internal.operators.OnSubscribeCreate$BufferEmitter.drain(OnSubscribeCreate.java:366)
	at rx.internal.operators.OnSubscribeCreate$BufferEmitter.onNext(OnSubscribeCreate.java:299)
	at com.gojuno.composer.FilesKt$tail$1$1.handle(Files.kt:15)
	at org.apache.commons.io.input.Tailer.readLines(Tailer.java:528)
	at org.apache.commons.io.input.Tailer.run(Tailer.java:470)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.gojuno.composer.InstrumentationEntry.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:118)
	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:190)
	at rx.internal.operators.OperatorScan$3.onNext(OperatorScan.java:140)
	... 16 more
```

**Comment by AntonRutkevich** [comment]
Just reproduced with almost identical stack trace.

**Comment by artem-zinnatullin** [comment]
Issue is pretty complicated, what we see is that `Tailer` from Apache Commons IO emits status code twice… 

And what we did for now is that we only match Instrumentation entries with different status codes into a test, this should just skip duplicated entries and in case of mismatch in number of tests in report and expected by `numtest` we'll crash so fix #20 should not create false-positive test run results.
